server/client.go

```diff
@@ -91,6 +92,12 @@ type clientOpts struct {
 	Version       string `json:"version"`
 }
 
+// Event data published when a client connects or disconnects
+type clientConnEvent struct {
```

Comment:
In the JSON, I think we at least need the client name to provide a meaningful way to tie an application to a client connection.  Maybe drop IP and Reason?